syft/file/contents_cataloger.go

```diff
@@ -0,0 +1,68 @@
+package file
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/source"
+)
+
+type ContentsCataloger struct {
+	globs              []string
+	skipFilesAboveSize int64
+}
+
+func NewContentsCataloger(globs []string, skipFilesAboveSize int64) (*ContentsCataloger, error) {
+	return &ContentsCataloger{
+		globs:              globs,
+		skipFilesAboveSize: skipFilesAboveSize,
+	}, nil
+}
+
+func (i *ContentsCataloger) Catalog(resolver source.FileResolver) (map[source.Location]string, error) {
+	results := make(map[source.Location]string)
+	var locations []source.Location
+
+	locations, err := resolver.FilesByGlob(i.globs...)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, location := range locations {
+		metadata, err := resolver.FileMetadataByLocation(location)
+		if err != nil {
+			return nil, err
+		}
+
+		if i.skipFilesAboveSize > 0 && metadata.Size > i.skipFilesAboveSize {
+			continue
+		}
+
+		result, err := i.catalogLocation(resolver, location)
+		if err != nil {
+			return nil, err
+		}
+		results[location] = result
+	}
+	log.Debugf("file contents cataloger processed %d files", len(results))
+
+	return results, nil
+}
+
+func (i *ContentsCataloger) catalogLocation(resolver source.FileResolver, location source.Location) (string, error) {
+	contentReader, err := resolver.FileContentsByLocation(location)
+	if err != nil {
+		return "", err
+	}
+	defer contentReader.Close()
+
+	buf := &bytes.Buffer{}
+	if _, err = io.Copy(base64.NewEncoder(base64.StdEncoding, buf), contentReader); err != nil {
+		return "", fmt.Errorf("unable to observe contents of %+v: %+v", location.RealPath, err)
```

Comment:
Nit: Do we still normally prefer `%w` for errors?

Reply:
yup yup, I'll adjust